qt/python/mantidqt/widgets/algorithmselector/widget.py

```diff
@@ -37,6 +37,19 @@ def get_name_and_version_from_item_label(item_label):
     return None
 
 
+class AlgorithmTreeWidget(QTreeWidget):
+
+    def __init__(self, parent=None):
+        super(AlgorithmTreeWidget, self).__init__(parent)
+        self.parent = parent
+
+    def mouseDoubleClickEvent(self, mouse_event):
+        if mouse_event.button() == Qt.LeftButton:
+            if not self.selectedItems()[0].child(0):
+                self.parent.execute_algorithm()
+            super(AlgorithmTreeWidget, self).mouseDoubleClickEvent(mouse_event)
```

Comment:
Should the `super()...` be one scope block higher than what it is (i.e. one less level of indentation)?

As it is now it will only call up if the left mouse button is the one triggering the double click event. From my understanding the call up should happen in every case you are not handling in this function.

Reply:
It seemed natural to me that the double-click with the right or centre button shouldn't expand the tree/open the run dialog, so I kept it inside the `if LeftButton` scope . But checking MantidPlot, the behaviour is that a double-click with any button performs the same action. So I've just removed the `if LeftButton` statement in the next commit,

Reply:
Yes, I agree that it should be left click only. My point was that calls to the parents event handler were dependent on the event being triggered by a left click.

Now that I think about it I can't remember if the parents event handler should always be called, or only called when you do not handle the event. @martyngigg would probably know.

Reply:
I agree it should be left-click only too. I don't think we should use all of the buttons. The event handler would only need to be called for the base if there was anything for it do and we actually want that behaviour to happen also. In this case I think we want to restrict that only left clicks do anything so not passing the event on is okay in the other cases.